### ISSUE TITLE
Failed to get private ssh key

### ISSUE THREAD

**Issue by mbrung** [codenvy/codenvy#2030]
Fairly often we get an error while trying to `git pull` from our Codenvy workspace: `Failed to get private ssh key. You can create a new SSH key pair in Profile->Preferences->SSH->VCS.`.  I understand what is being asked of me, but have already done so...  We have both set a private SSH key in the workspace configuration, and created a host in `Profile > Preferences > SSH > VCS` for our Git server.  Even after setting an SSH key pair, the problem remains.

When this happens I have to revert using Git in a terminal (with a `id_rsa` file manually stored in `~/.ssh/`).

Is there anything I could try to unblock the situation?

**Reproduction Steps:**

Unfortunately I am not able to consistently reproduce the problem from a blank workspace.  I can share an existing workspace that displays the problem if this could help.

**Codenvy version:**   codenvy.io 5.6.0

**Comment by eivantsov** [comment]
@mbrung hm.. it should not matter what workspace you use, since ssh key is stored in user prefs. What git hosting provider do you use?

**Comment by mbrung** [comment]
We use @planio as our hosting provider.

**Comment by planio** [comment]
Any way we can help to debug this?

**Comment by eivantsov** [comment]
I'll create an account and give it a try myself.

**Comment by eivantsov** [comment]
@mbrung @planio this is how I managed to import a project using SSH Git URL:

1. In Codenvy, go to Profile > Preferences > VCS > Generate Key:

hostname: `$myDomain.plan.io`

2. Copy the key
3. Save it at `https://$myDomain.plan.io/public_keys`
4. Import a sample plan.io project using ssh git url

![image](https://cloud.githubusercontent.com/assets/5337267/24643974/b29e46e8-1919-11e7-9be0-71cc67324829.png)

**Comment by mbrung** [comment]
@eivantsov absolutely, and I got it to work as well.

The problem lies further down the line; at some point, when trying to `git pull` from the UI the SSH key stored in the user preferences is no longer picked up, and I get the attached error, despite the key being present.

<img width="646" alt="screen shot 2017-04-04 at 08 50 09" src="https://cloud.githubusercontent.com/assets/10260/24644426/d318ba80-1913-11e7-8afd-b5f142df56c5.png">
<img width="367" alt="screen shot 2017-04-04 at 08 50 20" src="https://cloud.githubusercontent.com/assets/10260/24644427/d341b480-1913-11e7-9464-899f99036ce9.png">

**Comment by yeah** [comment]
Not sure if Codenvy supports it, but Planio also accepts [Git connections via HTTPS](https://support.plan.io/news/114). Maybe that's an alternative?

**Comment by eivantsov** [comment]
> At some point

Well, at the moment I am unable to reproduce it. I started and stopped my workspace and tried to pull.

When logged in, can you see your key when you GET in browser `https://codenvy.io/api/ssh/vcs`?

**Comment by mbrung** [comment]
I see my key (removed for obvious reasons ;-) 
```
[{"name":"ajileso.plan.io","publicKey":"REMOVED","service":"vcs","privateKey":"REMOVED","links":[{"method":"GET","parameters":[],"produces":"application/json","rel":"get pair","href":"https://codenvy.io/api/ssh/vcs"},{"method":"DELETE","parameters":[],"produces":"application/json","rel":"remove pair","href":"https://codenvy.io/api/ssh/vcs"},{"method":"GET","parameters":[],"produces":"application/json","rel":"get pair","href":"https://codenvy.io/api/ssh/vcs/find"}]}]
```

**Comment by eivantsov** [comment]
Tried to pull from repo I created and imported yesterday - no issues found. Perhaps, you may want to regenerate and re-upload key and see how it goes from there?

**Comment by mbrung** [comment]
Absolutely!  Have just done so (and have done so many times before), and got the result shown above.  Would you like access to the workspace directly? 

Perhaps unrelated, I opened the debugger in Safari, and noticed the following error:
```
[Error] Failed to load resource: the server responded with a status of 500 (Internal Server Error) (pull, line 0)
https://codenvy.io/33495_node7.codenvy.io/api/git/pull?projectPath=/webapp
```

**Comment by eivantsov** [comment]
@mbrung are you saying that it does not work at all in your case? Or works from time to time and and some point the key becomes invalid?

**Comment by mbrung** [comment]
@eivantsov I have seen it work in the past (some time in March, sorry I can't be any more precise).  My attempts in recent days have been unsuccessful, so I'd say it just does not work in our case.

**Comment by eivantsov** [comment]
Can you re-import your project and give it a shot?

**Comment by mbrung** [comment]
I am unable to do it at the moment, but will reimport tonight.

**Comment by mbrung** [comment]
@eivantsov actually I cannot reimport the project, I get the same error: "can't store ssh key.  Unable to get private ssh key".  Could there be something wrong with the workspace itself?

**Comment by eivantsov** [comment]
Delete an existing key. Generate a new one, save it to your plan.io account. This is how it worked for me. And yes, try a different workspace.

**Comment by mbrung** [comment]
> Delete an existing key. Generate a new one, save it to your plan.io account. This is how it worked for me

I just did.  Same result.  I trust this worked out for you, given this is how I started myself.  

> And yes, try a different workspace.

As it turns out this is going to prove difficult: I need extra RAM to do it, and the subscription appears to be permanent (the UI offers no way to decrease the quantity of RAM).

**Comment by eivantsov** [comment]
Again, I used a different account, and went through all the steps once again. I can import my project. Anything specific that you may do that differs from my flow?

**Comment by eivantsov** [comment]
@mbrung Just tried to clone git@codenvy123.plan.io:codenvy123/website-redesign.git into a new workspace. codenvy.io has been updated a few times since we discussed the problem last time. No issues found.

**Comment by eivantsov** [comment]
@mbrung is this still an issue for you?

**Comment by TylerJewell** [comment]
Closed due to inactivity.
